### PR TITLE
Kosta branch

### PR DESCRIPTION
**Pull request by Constantine19**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/IntSight/Integrations/IntSight/IntSight.py [hunk 1]
```diff
@@ -701,21 +703,181 @@ def search_for_ioc():
...
+                        'File': hash_info
+                    },
+                    'Contents': response,
+                    'HumanReadable': tableToMarkdown('IOC Enrichment', ioc_readable),
+                    'ContentsFormat': formats['json']
+                }
+            )
+            break
+        elif status == 'Queued' or status == 'InProgress':
+            time.sleep(2)
```
Comment (guykeller) [inline Packs/IntSight/Integrations/IntSight/IntSight.py:870]: Could you please explain why this is needed?
Usage of `sleep` is strongly discouraged, using polling is usually the better option

**Review by guykeller — COMMENTED** [review]
Overall looks good, I'd love to schedule a demo and view the changes live

**Review by guykeller — APPROVED** [review]

File: Packs/IntSight/Integrations/IntSight/IntSight.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 from CommonServerPython import *
 
-reload(sys)
-sys.setdefaultencoding('utf-8')  # pylint: disable=E1101
+# import sys
+import os
+# import importlib
 
+# importlib.reload(sys)
```
Comment (guykeller) [inline Packs/IntSight/Integrations/IntSight/IntSight.py:7]: Commented these lines out as this is a Python 2 integration, and also wasn't sure why they are necessary.
Can you explain or may I delete them?

**Review by guykeller — CHANGES_REQUESTED** [review]
Apologies it took some time to fix the build - we've been having some infrastructure issues.
Added two final notes, and fixed all other issues.
As for your local success VS build failures - it seems some of the code wasn't compatible with Python 2, which is the docker image the integration runs on
Once we fix these notes we should be good to merge

**Comment by guykeller** [comment]
Hi @Constantine19 ,
Just making sure you've seen my latest comments - I'd love to help you make them or answer any questions you might have

File: Packs/IntSight/Integrations/IntSight/IntSight.py [hunk 3]
```diff
@@ -979,6 +1144,8 @@ def test_module():
         update_ioc_blocklist_status()
     elif demisto.command() == 'intsights-close-alert':
         close_alert()
+    elif demisto.command() == 'intsights-test-action':
+        pass
```
[the remark this answers is not in the document]
Comment (Constantine19) [reply Packs/IntSight/Integrations/IntSight/IntSight.py:1148]: This is actually needed for the unit tests in the mocker. Please take a look at the file `IntSight_test.py` where we implemented the patch for this function.

File: Packs/IntSight/Integrations/IntSight/IntSight.py (hunk 2 above)
Comment (Constantine19) [reply Packs/IntSight/Integrations/IntSight/IntSight.py:7]: These libraries were included by Demisto back in the days. We tried not to touch anything that was there already. It is completely okay to remove them if that doesn't cause any issue with the overall integration

Comment (Constantine19) [reply Packs/IntSight/Integrations/IntSight/IntSight.py:7]: As for the `time.sleep()`:

I looked at the documentation and it looks like the polling has to be implemented on the playbook level (please correct me if I am wrong). Is there a way to implement in on a code level? If yes, then how? Thanks

**Comment by Constantine19** [comment]
Hi @guykeller , thanks for your inputs! Added my comments

File: Packs/IntSight/Integrations/IntSight/IntSight.py (hunk 2 above)
Comment (guykeller) [reply Packs/IntSight/Integrations/IntSight/IntSight.py:7]: It should actually be implemented at the code level.
You need to create one command that allows for requesting for enrichment, and one command that queries the status and updates the context with it.
Then, we get the results of the enrichment, and update the context.

**Comment by Constantine19** [comment]
Hey @guykeller , please look at this commit https://github.com/demisto/content/commit/efd16cec769be151184b07144270c73a18d986a5

We implemented the generic poll as requested. We did a relevant test on the playbook level and it looks good. Now, because this overall change requires customer to implement a gen polling, what is the best way to to tell them that?

File: Packs/IntSight/pack_metadata.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "name": "IntSights",
     "description": "Use IntSights to manage and mitigate threats.",
     "support": "xsoar",
```
Comment (bakatzir) [inline Packs/IntSight/pack_metadata.json:4]: is it going to stay xsoar supported? or are we changing ownership?

Comment (guykeller) [reply Packs/IntSight/pack_metadata.json:4]: Currently XSOAR supported

**Comment by guykeller** [comment]
@Constantine19 Code looks good, great job!
I'd love it if you could show a quick demo of it, possibly in a simple generic polling scenario. Please let me know when is a good time.

As for your question - your customer can look at the `HelloWorld_scan` playbook as a reference for creating such a playbook.

**Comment by guykeller** [comment]
@Constantine19 Code looks good, great job!
I'd love it if you could show a quick demo of it, possibly in a simple generic polling scenario. Please let me know when is a good time.

Also, it appears you have some conflicts and build errors - I suggest you pull from your base branch and resolve them. Let me know if you need my help with that.

As for your question - your customer can look at the HelloWorld_scan playbook as a reference for creating such a playbook.

**Comment by guykeller** [comment]
@Constantine19 Need to fix the mock issue, as we discussed on our session, and also bump the version to 1.0.5 (it appears by the error)

**Comment by Constantine19** [comment]
@guykeller , I have fixed the unit tests for Python 2 and bumped up the version. Please review and merge accordingly. Thank you very much for your help!

File: Packs/IntSight/Integrations/IntSight/IntSight.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (guykeller) [reply Packs/IntSight/Integrations/IntSight/IntSight.py:1150]: Last note - this still needs to be removed. We can't have phantom commands in the code.
You can look at how other integrations implemented tests without using such measures

File: Packs/IntSight/ReleaseNotes/1_0_5.md [hunk 5]
```diff
@@ -2,3 +2,4 @@
 #### Integrations
 ##### IntSights
 - Fixed display names for various integration parameters.
+- Added support for IOC enrichment
```
Comment (guykeller) [inline Packs/IntSight/ReleaseNotes/1_0_5.md:5]:
```suggestion
- Added Added the ***intsights-request-ioc-enrichment*** command which can be used for IOC enrichment.
```

**Review by guykeller — APPROVED** [review]
Approving - just remove the dummy command and change the tests accordingly please, and we're good to go

**Comment by Constantine19** [comment]
Hey @guykeller ,

Thanks for the approval! 

As for the dummy command:
I understand that there are strict guidelines on the way of writing an integration properly "by the book". We are currently working on passing this integration ownership to a team that will rewrite everything the way PaloAlto expects. This particular change is for one mutual customer of ours only. Hence, we are trying to push this asap. Your concern about the dummy command will certainly be addressed in the future release of our next generation IntSights app. 
The unit tests are currently passing on every IntSights command confirming that our Integration is working as expected. We, as a company, would really appreciate you merging this PR the way it is now. Future generation app will be written "by the book" as agreed with your team.

**Pull request merged**